src/pages/projects/common/components/CreateProjectModal.tsx

```diff
@@ -0,0 +1,229 @@
+  setVisible: Dispatch<SetStateAction<boolean>>;
+  onProjectCreated: (project: Project) => unknown;
+}
+
+export function CreateProjectModal(props: Props) {
+  const [t] = useTranslation();
+  const queryClient = useQueryClient();
+
+  const { data: blankProject } = useBlankProjectQuery();
+
+  const [project, setProject] = useState<Project>();
+  const [errors, setErrors] = useState<ValidationBag>();
+  const [isFormBusy, setIsFormBusy] = useState<boolean>(false);
+
+  const company = useCurrentCompany();
+  const clientResolver = useClientResolver();
+
+  const handleChange = (
+    property: keyof Project,
+    value: Project[typeof property]
+  ) => {
+    setErrors(undefined);
+    setProject((project) => project && { ...project, [property]: value });
+  };
+
+  useEffect(() => {
+    if (blankProject) {
+      setProject({
+        ...blankProject,
+        task_rate: company?.settings.default_task_rate || 0,
+      });
+    }
+  }, [blankProject]);
+
+  useEffect(() => {
+    if (project?.client_id && project.client_id.length > 1) {
+      clientResolver.find(project.client_id).then((client) => {
+        if (client.settings.default_task_rate) {
+          handleChange('task_rate', client.settings.default_task_rate);
+        }
+      });
+    }
+  }, [project?.client_id]);
+
+  const handleSave = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!isFormBusy) {
+      toast.processing();
+      setErrors(undefined);
+      setIsFormBusy(true);
+
+      request('POST', endpoint('/api/v1/projects'), project)
+        .then((response: GenericSingleResourceResponse<Project>) => {
+          toast.success('created_project');
+
+          queryClient.invalidateQueries('/api/v1/projects');
+
+          window.dispatchEvent(
+            new CustomEvent('invalidate.combobox.queries', {
+              detail: {
+                url: endpoint('/api/v1/projects'),
+              },
+            })
+          );
+
+          props.onProjectCreated(response.data.data);
+
+          setProject(blankProject);
+          props.setVisible(false);
+        })
+        .catch((error: AxiosError<ValidationBag>) => {
+          if (error.response?.status === 422) {
+            toast.dismiss();
+            setErrors(error.response.data);
+          } else {
+            console.error(error);
+            toast.error();
+          }
+        })
+        .finally(() => setIsFormBusy(false));
+    }
+  };
+
+  return (
+    <Modal
+      title={t('new_project')}
+      size="regular"
+      visible={props.visible}
+      onClose={() => {
+        setProject(blankProject);
+        props.setVisible(false);
+      }}
+    >
+      {project ? (
+        <>
+          <Element leftSide={t('project_name')}>
+            <InputField
+              value={project.name}
+              onValueChange={(value) => handleChange('name', value)}
+              errorMessage={errors?.errors.name}
+            />
+          </Element>
```

Comment:
`<Element>`  is meant for usage in cards, not modals. This way we end up with double padding, causing elements to not align.

Reply:
In that case, I removed the `Element` component and the name of the filed is managed with the `label` and `inputLabel` props. But the whole UI doesn't look good with only one column IMO. So, I separated it into two columns of 4 fields. The description has been updated with a screenshot of the new UI. Let me know your thoughts.